InnaFeature/Steps/WidgetsSteps.cs

```diff
@@ -0,0 +1,198 @@
+            widgets.MultipleColorNamesField.SendKeys(letter);
+        }
+
+        [When(@"User navigates to ""([^""]*)"" section")]
+        public void UserNavigatesToAutoCompleteSection(string sectionName)
+        {
+            basePage.NavigateToTheSection(sectionName);
+        }
+
+        [When(@"User adds the colors Red, Yellow, Green, Blue, and Purple to the ""([^""]*)"" field")]
+        public void WhenUserAddsTheColorsRedYellowGreenBlueAndPurpleToTheField()
+        {
+            widgets.MultipleColorNamesField.Click();
+            var actions = new Actions(browserHelper.WebDriver);
+
+
+            foreach (var color in ColorsInField)
+            {
+                widgets.MultipleColorNamesField.SendKeys(color);
+                Thread.Sleep(500);
+                widgets.MultipleColorNamesField.SendKeys(Keys.Enter);
+                Thread.Sleep(500);
+            }
+        }
+
+        [When(@"User removes ""([^""]*)"" and ""([^""]*)""")]
+        public void WhenUserRemovesYellowAndPurple(string color1, string color2)
+        {
+            widgets.RemoveColor(color1).Click();
+            widgets.RemoveColor(color2).Click();
+        }
+
+        [When(@"User navigates to ""([^""]*)"" section")]
+        public void WhenUserNavigatesToProgressBarSection(string sectionName)
+        {
+            IJavaScriptExecutor js = (IJavaScriptExecutor)browserHelper.WebDriver;
+            js.ExecuteScript("arguments[0].scrollIntoView(true);", sectionName);
+            basePage.NavigateToTheSection(sectionName);
+        }
+
+        public int GetProgressValue(IWebElement progressBar)
+        {
+            string progressText = progressBar.Text;
+            string progressValue = progressText.Trim('%');
+            if (int.TryParse(progressValue, out int value))
+            {
+                return value;
+            }
+            else
+            {
+                return -1;
+            }
+        }
+
+        [When(@"User clicks Start and wait until the progress reaches (.*)%")]
+        public void WhenUserClicksStartAndWaitUntilTheProgressReaches(int targetProgress)
+        {
+            widgets.StartButton.Click();
+            int currentProgress = 0;
+            int maxWaitTimeInSeconds = 10;
+            int waitTime = 0;
+            int pollingInterval = 1000;
+
+            while (currentProgress < targetProgress && waitTime < maxWaitTimeInSeconds * 1000)
+            {
+                currentProgress = GetProgressValue(widgets.ProgressBar);
+                Thread.Sleep(pollingInterval);
+                waitTime += pollingInterval;
+
+            }
+            if (currentProgress >= targetProgress)
+            {
+                Console.WriteLine($"Progress reached {targetProgress}%");
+            }
+            else
+            {
+                Console.WriteLine($"Progress did not reach {targetProgress}% within {maxWaitTimeInSeconds} seconds");
+            }
```

Comment:
Basically, you don't need to do WriteLine in specflow tests. Applicable to all usages within test project